resources/js/app/components/drop-upload.js

```diff
@@ -101,19 +101,32 @@ export default {
             this.uploadFiles(files);
         },
 
-        uploadFiles(files) {
+        async uploadFiles(files) {
             this.$el.classList.remove('dragover');
-            ([...files]).forEach(f => {
-                if (this.$helpers.checkMaxFileSize(f) === false) {
+            // show all files in view
+            [...files].forEach(file => this.setProgressInfo(file));
+            const [uniqueFiles, duplicatesFiles] = this.filterFiles([...files]);
+            uniqueFiles.forEach((file) => {
+                // skip file not allowed by size and remove it from view
+                if (this.$helpers.checkMaxFileSize(file) === false) {
+                    this.removeProgressItem(file);
                     return;
                 }
-                this.upload(f)
-                    .then((object) => {
-                        this.$emit('new-relations', [object]);
-                        this.removeProgressItem(f);
-                    });
 
+                this.upload(file).then((object) => this.uploadSuccessful(file, object));
             });
+
+            // use queue for file with same name
+            for (const file of duplicatesFiles) {
+                // skip file not allowed by size and remove it from view
+                if (this.$helpers.checkMaxFileSize(file) === false) {
+                    this.removeProgressItem(file);
+                    return;
```

Comment:
I expected a `continue` here... otherwise this exits the function and other files in `duplicatedFiles` are not processed for upload